docs/upgrading-2.0.rst

```diff
@@ -0,0 +1,74 @@
+deprecated and it is recommended to upgrade from the legacy systems.
+
+.. _upgrading_auth:
+
+Upgrading Authentication/Authorization
+--------------------------------------
+
+The authentication and authorization policies of Pyramid 1.x have been merged
+into a single :term:`security policy` in Pyramid 2.0.  Authentication and
+authorization policies can still be used and will continue to function
+normally, however they have been deprecated and support may be removed in
+upcoming versions.
+
+The new security policy should implement
+:interface:`pyramid.interfaces.ISecurityPolicy` and can be set via the
+``security_policy`` argument of :class:`pyramid.config.Configurator` or
+:meth:`pyramid.config.Configurator.set_security_policy`.
+
+The new security policy merges ``unauthenticated_userid`` and
+``authenticated_userid`` into an :term:`identity` object.  This object can be
+of any shape, such as a simple ID string or an ORM object, but should   The
+identity can be accessed via
+:prop:`pyramid.request.Request.authenticated_identity`.
+
+The concept of :term:`principals <principal>` has been removed; the
+``permits`` method is passed an identity object.  This change gives much more
+flexibility in authorization implementations, especially those that do not
+match the ACL pattern.  If you were previously using
+:class:`pyramid.authorization.ACLAuthorizationPolicy`, you can achieve the same
+results by writing your own ``permits`` method using
+:class:`pyraid.authorization.ACLHelper`.  For more details on implementing an
+ACL, see :ref:`assigning_acls`.
+
+Pyramid does not provide any built-in security policies.  Similiar
+functionality of the authentication and authorization policies is now provided
+by helpers, which can be utilized to easily implement your own security policy.
+The functionality of the legacy authencation policies roughly correspond to the
+following helpers
+
+* :class:`pyramid.authentication.SessionAuthenticationPolicy`:
+  :class:`pyramid.authentication.SessionAuthenticationHelper`
+* :class:`pyramid.authentication.AuthTktAuthenticationPolicy`:
+  :class:`pyramid.authentication.AuthTktCookieHelper`
+* :class:`pyramid.authentication.BasicAuthAuthenticationPolicy`:
+  Use :func:`pyramid.authentication.extract_http_basic_credentials` to retrieve
+  credentials.
+* :class:`pyramid.authentication.RemoteUserAuthenticationPolicy`:
+  ``REMOTE_USER`` can be accessed with ``request.environ.get('REMOTE_USER')``.
+* :class:`pyramid.authentication.RepozeWho1AuthenticationPolicy`:
+  No equivalent.
+
+For further documentation on implementing security policies, see
+:ref:`writing_security_policy`.
+
+Behavior of the Legacy System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Legacy authentication and authorization policies will continue to function as
+normal, as well as all related :class:`pyramid.request.Request` properties.
+The new :prop:`pyramid.request.Request.authenticated_identity` property will
```

Comment:
`:prop:` is a new one to me! It looks like `:prop:` requires Sphinx 2.1+. If that assumption is true, then we would need to bump the version minimum version in `setup.py` accordingly.

It would also be good to verify that it builds on RTD, too, before merging this PR. To do so, you can setup your fork on RTD and build the specific branch.

Reply:
Whoops, didn't realize it was 2.1+.  As of a couple months ago, RTD was running Sphinx 1.8, so I think we'll have to get rid of it.

What's the recommended way to reference properties in older versions of Sphinx?

Reply:
If memory serves me correctly, we can force RTD to use a more current version of Sphinx via the use of:

* modify `setup.py`
* invoke via `rtd.txt`'s `-e .[docs]`, which in turn is invoked by...
* configuring RTD under _Advanced Settings_ to use `rtd.txt` as a requirements file and install into a virtual environment (see screenshot).
![Google Chrome007](https://user-images.githubusercontent.com/102112/59152051-da0e3a80-89f1-11e9-837a-1b2cf35986dc.png)

I can grant you access to the RTD configuration as an admin if you need to compare setup if you provide me with your RTD username. I might have omitted a detail, but I think that covers it all.

Reply:
Looks like `:attr:` does what I'm looking for and that works in 1.8.  I'm not even sure where I found `:prop:`, I don't see it in the Sphinx documentation.

IMO best to stick with the RTD Sphinx version if we can help it.